### PR TITLE
Contributing guideline added

### PR DESCRIPTION
**Pull request by mrturkmen06**
Includes some information about code review, conventions and contribution rules.

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+## Conventions 
+
+Fork the repo and make your changes on your fork in a feature branch. Haaukins developments is based on [Git Feature Branch Workflow](https://www.atlassian.com/git/tutorials/comparing-workflows/feature-branch-workflow)
+
+- For bug fixes and feature requests, the name of branch that you will work on should follow the convention [XXXX]-something, XXXX refers to issue number. 
+
+- There should be a naming convention (function names and variable names should follow defined pattern such as getName(private) or GetName (public).
+
+- Having clean formatted codes are always easy to review, hence the changes on branch require to have `gofmt` to format the code.
```
Comment (kidmose) [inline .github/CONTRIBUTING.md]: - Use `gofmt` to ensure code formatting that is easy to read.

Comment (kidmose) [reply .github/CONTRIBUTING.md]: ... a suggestion for a more concise wording ("Rule"  comes first, followed by reasoning)

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+- Have descriptive comments on changes.
+
+## For Maintainers
+
+### Code Review Process 
+
+- All changes to the master branch must be through Pull Requests, and all PRs must be approved by at least one maintainer, who must be different from the creator of the PR. 
+
+- Pull requests should be made into version branch instead of master, since latest version will be identical with master branch, having pull request to version branch will bump version according to changes.
+
```
Comment (kidmose) [inline .github/CONTRIBUTING.md]: Should the PR made by a contributor point to the version branch or should the maintainer merge the PR into the version branch? (I guess only the maintainer knows this)

**Comment by kidmose** [comment]
It's a bit unclear what belongs in `## Conventions` and what belongs in `## For maintainers` - both are for instance concerned with the workflow. Can you try and clarify this, perhaps by collecting these in a dedicated `## Workflow` section, or perhaps checking that it all is found under Conventions?

See (My attempt at) inline comments too.

File: .github/CONTRIBUTING.md (hunk 2 above)
Comment (mrturkmen06) [reply .github/CONTRIBUTING.md]: I am a little bit confused for the initial question, 

> Should the PR made by a contributor so it points to the right version branch or should the maintainer merge the PR into the version branch?

Making PR and merging the PR are two different things, so a contributor will open a pull request to version branch which is identical to latest version. For example, let's say, 1.6.3 is latest version branch (identical with master), and ideally `( - automated release script has created the next patched version called 1.6.4 (it is identical to 1.6.3 )- )` so when 1.6.4 branch is created, next developments should take place based on 1.6.3, and when pull requests created the base target should be 1.6.4, so 1.6.4 will contain the changes which are made from 1.6.3.

Comment (kidmose) [reply .github/CONTRIBUTING.md]: Sorry if my question was not clear, but I think you got it anyway. At least you answered what I wanted to ask :)

I don't think the point you just made is clear from CONTRIBUTING.
Do you mind having a second look at it, if you indeed think it is.
Modify if relevant, or indicate if you think it's fine - then I'm happy too :+1:

Comment (kidmose) [reply .github/CONTRIBUTING.md]: nvm. You already did that..

**Comment by eyJhb** [comment]
Little late now @mrturkmen06  and @kidmose, but I would have suggested to use `goimports` instead of `go fmt`, and also force the use of `go vet`, as it catches annoyances. :)

**Comment by mrturkmen06** [comment]
Although it is late :), we will figure out to put your suggestions in some point (before releasing new version). Thank you for the feedback.

**Comment by kidmose** [comment]
@eyJhb Sure! Can you capture the _why_ and the suggestion for _what_ in an issue, and we'll prioritise it!
